modules/s1/main.tf

```diff
@@ -0,0 +1,570 @@
+provider "aws" {
+  region = var.region
+}
+
+locals {
+  api_domain_name = join(".", [var.api_subdomain, var.domain_name])
```

Comment:
the actual application will run on `subdomain.domain.name`, due to SSL / CloudFlare complexities we cannot have multi-level subdomains... the api will be on (generally) `subdomain-api.domain.name` - does this affect anything?

Reply:
I don't think so, that means TF will need `domain.name` as domain_name var and `subdomain-api` as subdomain-api var